### PR TITLE
Feature survey json converter

### LINKED ISSUE #1151: Implement new push/push answer json structure

### PR DESCRIPTION
**Pull request by manuelplazaspalacio**
Close #1151

### CODE REVIEW

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/push/PushUseCase.java [hunk 1]
```diff
@@ -75,50 +78,13 @@ public void execute(final Callback callback) {
...
-                        } else {
-                            notifyReOpenOrgUnit();
-                        }
-                    }
-                });
+        mPushUseCaseStrategy.run();
     }
 
-    private boolean isOrgUnitBanned() {
+    public boolean isOrgUnitBanned() throws NetworkException, ApiCallException {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/usecase/push/PushUseCase.java]: is dangerous refactor to public this methods for be invokes from strategy but can be invoked from presentation layer also, in PushUseCase only execute method should be public then I think is better refactor for move this methods to strategy base

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/push/PushUseCase.java [hunk 2]
```diff
@@ -44,7 +45,7 @@
         void onApiCallError(ApiCallException e);
     }
 
-    private IPushController mPushController;
+    public IPushController mPushController;
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/usecase/push/PushUseCase.java]: is dangerous refactor to public pushController for be invokes from strategy but can be invoked from presentation layer also, in PushUseCase only execute method should be public then I think is better refactor for pass pushcontroller as argument in strategy constructor

File: app/src/main/java/org/eyeseetea/malariacare/domain/usecase/push/PushUseCase.java [hunk 3]
```diff
@@ -196,6 +162,14 @@ private void banOrgUnit() {
         }
     }
 
+    public IPushController getPushController() {
```
Comment (xurxodev) [inline app/src/main/java/org/eyeseetea/malariacare/domain/usecase/push/PushUseCase.java]: If we pass PushController and OrganisationUnitRepository as arguments to strategy constructor this getters is not necessary

**Comment by manuelplazaspalacio** [comment]
@xurxodev Changes done.

**Comment by manuelplazaspalacio** [comment]
@xurxodev  put the needed methods in the PushUseCaseStartegy and removed it from the PushUseCase

**Pull request merged**